Modules/signalmodule.c

```diff
@@ -1737,20 +1726,25 @@ _PyErr_CheckSignals(void)
 }
 
 
-/* Simulate the effect of a signal.SIGINT signal arriving. The next time
-   PyErr_CheckSignals is called,  the Python SIGINT signal handler will be
-   raised.
+/* Simulate the effect of a signal arriving. The next time PyErr_CheckSignals
+   is called,  the corresponding Python signal handler will be raised.
 
-   Missing signal handler for the SIGINT signal is silently ignored. */
+   Missing signal handler for the given signal number is silently ignored. */
 void
-PyErr_SetInterrupt(void)
+PyErr_SetInterruptEx(int signum)
 {
-    if ((Handlers[SIGINT].func != IgnoreHandler) &&
-        (Handlers[SIGINT].func != DefaultHandler)) {
-        trip_signal(SIGINT);
+    if ((Handlers[signum].func != IgnoreHandler) &&
+        (Handlers[signum].func != DefaultHandler)) {
```

Comment:
Ensure that `signum` is less than `NSIG` before accessing `Handlers[signum]`.

Reply:
I would say that it's a precondition for calling this function. This is for expert-level C code. Though I could still test it and make the function a no-op if the signal number is out of bounds.

Reply:
That makes sense to me, but then let's make sure it's mentioned in the docs of `PyErr_SetInterruptEx`

Reply:
It's just basic validation of the argument of an API function, to get an error instead of crashing the process with segfault or access violation.

Reply:
Good idea, will do.